src/Link/index.tsx

```diff
@@ -143,22 +151,54 @@ export default class Link extends React.PureComponent<LinkProps, LinkState> {
     this.props.onMouseOut(this.props.linkData.source, this.props.linkData.target, evt);
   };
 
+  getLinkColor(): string[] | string | undefined| null  {
+    const { linkData, pathColorFunc } = this.props;
+    let color = pathColorFunc;
+    if (typeof pathColorFunc === 'function') {
+      color =  pathColorFunc(linkData)  as string | string[];
+    }
+
+    return color;
+  }
+
+
+  getStrokeValue(linkColor, gradientId): string {
+    if (Array.isArray(linkColor)) {
+      return `url(#${gradientId})`;
+    } else {
+      return linkColor;
+    }
+  }
+
   render() {
-    const { linkData } = this.props;
+    const { linkData, pathColorFunc } = this.props;
+    const gradientId = getId();
+    const linkColor = this.getLinkColor();
+    const [gradientStartColor, gradientEndColor] = Array.isArray(linkColor) ? linkColor : [undefined, undefined];
+
     return (
-      <path
-        ref={l => {
-          this.linkRef = l;
-        }}
-        style={{ ...this.state.initialStyle }}
-        className={this.getClassNames()}
-        d={this.drawPath()}
-        onClick={this.handleOnClick}
-        onMouseOver={this.handleOnMouseOver}
-        onMouseOut={this.handleOnMouseOut}
-        data-source-id={linkData.source.id}
-        data-target-id={linkData.target.id}
-      />
+      <>
+      {gradientStartColor && <defs>
```

Comment:
Consider extracting this to a method to be used like this `{this.getGradientDef(gradientStartColor, gradientEndColor)` (`linkData` can be extracted from the props inside `getGradientDef`)